config/routes.rb

```diff
@@ -345,14 +343,19 @@
     # While this is the start of a v1 API, this is currently not usable by outside developers as
     # getting a JWT token requires you to be logged in through the Website
     namespace :v1 do
-      namespace :registrations do
-        get '/register', to: 'registrations#show'
-        post '/register', to: 'registrations#create'
-        patch '/register', to: 'registrations#update'
-        patch '/bulk_update', to: 'registrations#bulk_update'
-        get '/:competition_id', to: 'registrations#list'
-        get '/:competition_id/admin', to: 'registrations#list_admin', as: :list_admin
-        get '/:competition_id/payment', to: 'registrations#payment_ticket', as: :payment_ticket
+      resources :competitions, only: [] do
+        resources :registrations, only: %i[index show create update], shallow: true do
+          member do
+            get 'payments', to: 'registrations#registration_payments'
```

Comment:
moving this here now means we want this part of the public API, which I am not sure of

Reply:
Eventually everything has to be part of v1 so that it can be served to Next. Is there a better place for this API, especially considering our long-term infrastructure plan?

Reply:
Good point with Next, I think it would make more sense behind an internal API. There will be dozens of routes that we want to expose in a different internal scope in the future when we are moving more and more pages to next. Even in registrations there are routes like the payment completion URL that don't live in v1 and I would argue shouldn't.

Reply:
Then what do we do with this route in this PR?

Reply:
keep it under `resources :registrations, only: %i[index update create edit destroy], shallow: true` ?

Reply:
That does not sound like a long-term solution :/